File: 0015-3sum/0015-3sum.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        ans = set()
        for i in range(len(nums)-2):
            target = 0 - nums[i]
            j,k = i+1,len(nums)-1
            while j < k:
                if nums[j]+nums[k] == target:
                    ans.add((nums[i],nums[j],nums[k]))
                    j+=1
                    k-=1
                elif nums[j]+nums[k] > target:
                    k-=1
                else:
                    j+=1
        return list(ans)

File: 0015-3sum/0015-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Set<List<Integer>> set = new HashSet<>();
        Arrays.sort(nums);
        for(int i=0;i<nums.length-2;i++){
            int target = 0 - nums[i],j = i+1,k = nums.length-1;
            while(j<k){
                if(nums[j]+nums[k]<target)
                    j++;
                else if(nums[j]+nums[k]>target)
                    k--;
                else{
                    set.add(new ArrayList<>(Arrays.asList(nums[i],nums[j],nums[k])));
                    j++;
                    k--;
                }
            }
        }
        return new ArrayList<>(set);
    }
}